doc/tutorial/search/index.md

```diff
@@ -0,0 +1,292 @@
+### Bi-FM-Index
+In contrast to the FM-Index, the Bidirectional Index (Bi-FM-Index) allows searches to be conducted in both directions, i.e. from right to left and left to right. Being able to start at any position of the query in a search improves the search performance drastically when allowing multiple errors. You can find more information on Bi-FM-Indicies [here](https://doi.org/10.1109/BIBM.2009.42).
+However, it also needs to store additional information to be bidirectional. You can think of the Bi-FM-Index as two FM-Indices, one for each direction, hence you need about 2 times more space for the Bi-FM-Index.<br>
+
+## Example
+
+Constructing a (Bi-)FM-Index is very simple:
+
+\include doc/tutorial/search/search_basic_index.cpp
+
+You can also index text collections (e.g. genomes with multiple chromosomes or protein databases):
+
+\snippet doc/tutorial/search/search_small_snippets.cpp text_collection
+
+The indices can also be stored and loaded from disk.<br>
+Both seqan3::fm_index::store and seqan3::fm_index::load return a `bool` indicating whether the operation was successful.
+
+\snippet doc/tutorial/search/search_small_snippets.cpp store
+
+\snippet doc/tutorial/search/search_small_snippets.cpp load
+
+Note that in contrast to the construction via a given `text`, the template cannot be deduced by the compiler when using the default constructor so you have to provide template arguments.
+<a name="assignment_create_index"></a>
+\assignment{Assignment 1}
+You are given the text
+\code
+dna4_vector text{"CGCTGTCTGAAGGATGAGTGTCAGCCAGTGTAACCCGATGAGCTACCCAGTAGTCGAACTGGGCCAGACAACCCGGCGCTAATGCACTCA"_dna4};
+\endcode
+Create a seqan3::fm_index over the reference, store the index and load the index into a new seqan3::fm_index object.
+Print whether the store and load operations were successful.
+\endassignment
+
+\solution
+\snippet doc/tutorial/search/search_solution1.cpp solution
+**Expected output:**
+```console
+Index stored successfully.
+Index loaded successfully.
+```
+\endsolution
+
+# Search
+
+Using an index, we can now conduct searches for a given query. In this part we will learn how to search exactly, allow substitutions and indels, and how to configure what kind of results we want, e.g. all results vs. only the best result.
+
+## Terminology
+### Search Scheme
+A search scheme is the manner of how a search is conducted, e.g. trying to match each position in the reference to the query.
+### Optimal Search Scheme
+The scheme that requires the least amount of queries in the index is called an *optimal search scheme*.
+
+SeqAn3 provides *optimal* search schemes for up to **4** errors for Bi-FM-Indices and provides sub-optimal but still efficient search schemes otherwise.
+You can read more about search schemes in [this paper](http://doi.org/10.1101/301085).
+
```

Comment:
Mhm, it's not that important since it happens in the background.

But it's maybe important to know that for more than 4 errors the search get's way slower since the scheme is not optimal anymore.

And it's something fancy we have, so we might as well advertise it? 😁

Reply:
> But it's maybe important to know that for more than 4 errors the search get's way slower since the scheme is not optimal anymore.

Who searches with 4 error anyway? :wink: 

> And it's something fancy we have, so we might as well advertise it?

Except that our readers don't understand the fanciness and just think "seqan is complicated" :expressionless:

Reply:
I agree with Hannes. Hopefully we will soon have a generic efficient solution for >4 errors :)

I implemented 01*0 seeds for my thesis formulated as search schemes for which I could make a PR after I finished work on the SDSL.

Reply:
To be more precise: If I read that something is optimal until 4 errors and not optimal after that, it sounds like it will misbehave like return worse/bad results or miss something after four errors. I would really not write this in the tutorial. It has its place in the API docs, but not here.